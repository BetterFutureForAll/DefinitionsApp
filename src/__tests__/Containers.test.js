import { render } from "@testing-library/react";
import Components from "../containers/ComponentBox";
import Dimensions from "../containers/DimensionBox";
import renderer from 'react-test-renderer';


describe('Containers separate data into their unique Dimensions and Components', () => {

  test('Creates Component-Box container with multiple Components', () => {

    let testProps = [
      {
     "indicator_number": "1",
     "dimension": "Basic Human Needs",
     "component": "Nutrition and Basic Medical Care",
     "indicator_name": "Infectious diseases ",
     "unit_of_measurement": "(DALYs/100,000)",
     "definition": "d",
     "source": "Institute for Health Metrics and Evaluation",
     "link": "http://ghdx.healthdata.org/gbd-results-tool",
     "notes": "The sum"
   },
   {
     "indicator_number": "2",
     "dimension": "Basic Human Needs",
     "component": "Nutrition and Basic Medical Care",
     "indicator_name": "Infectious diseases ",
     "unit_of_measurement": "(DALYs/100,000)",
     "definition": "d",
     "source": "Institute for Health Metrics and Evaluation",
     "link": "http://ghdx.healthdata.org/gbd-results-tool",
     "notes": "The sum"

   },
   {
     "indicator_number": "4",
     "dimension": "Basic Human Needs",
     "component": "test",
     "indicator_name": "Infectious diseases ",
     "unit_of_measurement": "(DALYs/100,000)",
     "definition": "d",
     "source": "Institute for Health Metrics and Evaluation",
     "link": "http://ghdx.healthdata.org/gbd-results-tool",
     "notes": "The sum"

   },
   {
     "indicator_number": "0",
     "dimension": "",
     "component": "",
     "indicator_name": "tester test",
     "unit_of_measurement": "(100,000)",
     "definition": "test",
     "source": "test",
     "link": "test",
     "notes": ""
   }
]
    let rawData = require('../assets/definitions.json');
    
    const tree = renderer.create(<Components data={rawData.definitionsArray}></Components>).toJSON();

    expect(tree).toMatchSnapshot();

  });

});