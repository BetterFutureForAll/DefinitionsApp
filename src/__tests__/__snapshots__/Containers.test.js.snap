// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers separate data into their unique Dimensions and Components Creates Component-Box container with multiple Components 1`] = `
<div
  className="component-box"
>
  <div
    className="component"
    id="Nutrition_and_Basic_Medical_Care"
  >
    <p>
      Do people have enough food to eat and are they receiving basic medical care? 
    </p>
    <ul>
      <li
        class="list-item"
        title="d"
      >
        Infectious diseases 
        <a
          class="citation"
          href="http://ghdx.healthdata.org/gbd-results-tool"
          rel="noopener noreferrer"
          target="_blank"
          title="Institute for Health Metrics and Evaluation"
        >
          ⓘ
        </a>
      </li>
      <li
        class="list-item"
        title="d"
      >
        Infectious diseases 
        <a
          class="citation"
          href="http://ghdx.healthdata.org/gbd-results-tool"
          rel="noopener noreferrer"
          target="_blank"
          title="Institute for Health Metrics and Evaluation"
        >
          ⓘ
        </a>
      </li>
    </ul>
    <img
      alt="Do people have enough food to eat and are they receiving basic medical care? "
      className="component-img"
      src="test-file"
    />
    <h5
      class="component-title"
      style="backgroundColor: rgb(225, 100, 100)"
    >
      Nutrition and Basic Medical Care
    </h5>
  </div>
  <div
    className="component"
    id="test"
  >
    <p />
    <ul>
      <li
        class="list-item"
        title="d"
      >
        Infectious diseases 
        <a
          class="citation"
          href="http://ghdx.healthdata.org/gbd-results-tool"
          rel="noopener noreferrer"
          target="_blank"
          title="Institute for Health Metrics and Evaluation"
        >
          ⓘ
        </a>
      </li>
    </ul>
    <img
      alt=""
      className="component-img"
      src=""
    />
    <h5
      class="component-title"
      style="backgroundColor: "
    >
      test
    </h5>
  </div>
</div>
`;
